src/main/java/com/fitpet/server/badge/application/service/BadgeCheckServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package com.fitpet.server.badge.application.service;
+
+import com.fitpet.server.badge.application.mapper.BadgeCheckMapper;
+import com.fitpet.server.badge.domain.entity.Badge;
+import com.fitpet.server.badge.domain.entity.BadgeCheck;
+import com.fitpet.server.badge.domain.exception.BadgeCheckNotFoundException;
+import com.fitpet.server.badge.domain.exception.BadgeNotFoundException;
+import com.fitpet.server.badge.domain.repository.BadgeCheckRepository;
+import com.fitpet.server.badge.domain.repository.BadgeRepository;
+import com.fitpet.server.badge.presentation.dto.BadgeCheckCreateRequest;
+import com.fitpet.server.badge.presentation.dto.BadgeCheckDto;
+import com.fitpet.server.user.domain.entity.User;
+import com.fitpet.server.user.domain.exception.UserNotFoundException;
+import com.fitpet.server.user.domain.repository.UserRepository;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class BadgeCheckServiceImpl implements BadgeCheckService {
+
+    private final BadgeRepository badgeRepository;
+    private final BadgeCheckRepository badgeCheckRepository;
+    private final BadgeCheckMapper badgeCheckMapper;
+    private final UserRepository userRepository;
+
+    @Override
+    public BadgeCheckDto assignBadge(Long userId, BadgeCheckCreateRequest request) {
+        log.info("[BadgeCheckService] 뱃지 부여 요청: userId={}, badgeId={}", userId, request.badgeId());
+        Badge badge = badgeRepository.findById(request.badgeId())
+                .orElseThrow(BadgeNotFoundException::new);
+        User user = userRepository.findById(userId)
+                .orElseThrow(UserNotFoundException::new);
+
+        BadgeCheck badgeCheck = badgeCheckRepository.findByUserIdAndBadgeId(userId, request.badgeId())
+                .orElseGet(() -> badgeCheckMapper.toEntity(user, badge));
+
+        BadgeCheck saved = badgeCheckRepository.save(badgeCheck);
```

Comment:
로직 잘 짰네요! 👀 
한 가지 제안 드리고 싶은 부분이 있습니다.

현재는 뱃지와 사용자를 먼저 저회한 후 뱃지 보유 유무를 체크하고 있습니다.
만약에 사용자가 이미 뱃지를 소유하고 있다면 불필요한 비용이 발생할 것으로 보입니다😅

`badgeCheckRepository.findByUserIdAndBadgeId` 조회를 맨 위로 올려서 중복 여부를 먼저 확인하는 순서로 변경하면 더 효율적일 것 같습니다!🫡

 

```suggestion
   BadgeCheck badgeCheck = badgeCheckRepository.findByUserIdAndBadgeId(userId, request.badgeId())
          .orElseGet(() -> {
              // 없을 때만 조회
              Badge badge = badgeRepository.findById(request.badgeId())
                      .orElseThrow(BadgeNotFoundException::new);
              User user = userRepository.findById(userId)
                      .orElseThrow(UserNotFoundException::new);
              return badgeCheckMapper.toEntity(user, badge);
          });
  
  BadgeCheck saved = badgeCheckRepository.save(badgeCheck);
```

Reply:
헉 좋은지적 감사합니다 ! 
반영했으니 확인 부탁드립니당 !